vgscollect/src/main/java/com/verygoodsecurity/vgscollect/util/extension/InputStream.kt

```diff
@@ -0,0 +1,15 @@
+package com.verygoodsecurity.vgscollect.util.extension
+
+import java.io.InputStream
+
+internal class NotEnoughMemoryException : Exception()
+
+@Throws(NotEnoughMemoryException::class)
+internal inline fun <R> InputStream.useIfMemoryEnough(block: (InputStream) -> R): R {
+    if (this.available() >= Runtime.getRuntime().freeMemory() / 2) {
```

Comment:
I think it should rely on the value which was set by user with `vgsForm.getFileProvider().resize(cacheSize)` method.